src/lib.rs

```diff
@@ -297,6 +297,28 @@ impl<P: AsRef<[u8]>, T: Transitions> Automaton<P> for AcAutomaton<P, T> {
     }
 }
 
+// `(0..256).map(|b| b as u8)` optimizes poorly in debug builds so
+// we use this small explicit iterator instead
+struct AllBytesIter(i32);
+impl Iterator for AllBytesIter {
+    type Item = u8;
+    #[inline]
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.0 <= 256 {
+            let b = self.0 as u8;
+            self.0 += 1;
+            Some(b)
+        } else {
+            None
+        }
+    }
+}
+
+#[inline]
+fn all_bytes() -> AllBytesIter {
+    AllBytesIter(0)
+}
```

Comment:
This seems a little weird to me. Why not:

```rust
impl AllBytesIter {
    fn new() -> AllBytesIter {
        AllBytesIter(0)
    }
}
```

Also, does the function need to be `#[inline]`?

Reply:
Fixed